demo/src/hooks/usePresence.ts

```diff
@@ -0,0 +1,58 @@
+import * as Chat from '@ably/chat';
+import { useEffect, useState } from 'react';
+import { useRoom } from './useRoom.ts';
+
+export const usePresence = () => {
+  const [presentUsers, setPresentUsers] = useState<{ clientId: string; status: string }[]>([]);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [present, setPresent] = useState<boolean>(false);
+  const { room } = useRoom();
+
+  useEffect(() => {
+    if (!room) return;
+    const { unsubscribe } = room.presence.subscribe(() => {
+      room.presence.get().then((members) => {
+        setPresentUsers(
+          members.map((member) => {
+            {
+              const data = member.data as { status: string };
+              return { clientId: member.clientId, status: data.status };
+            }
+          }),
+        );
+      });
+    });
+    setLoading(false);
+    return () => {
+      // cleanup after the component is unmounted
+      unsubscribe();
+    };
+  }, [room]);
+
+  const enterPresence = (data?: Chat.PresenceData) => {
+    room.presence.enter(data).then(() => {
```

Comment:
This Promise should be returned so that the outside user can see its result

Reply:
All the promises are now returned and handled by the outside user :)